hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -3654,14 +3654,15 @@ void setQuota(String src, long nsQuota, long ssQuota, StorageType type)
       checkSuperuserPrivilege(operationName, src);
     }
     try {
-      writeLock();
+      // Need to compute the curren space usage
+      writeLock(FSNamesystemLockMode.GLOBAL);
```

Comment:
@ZanderXu Sir. I can not understand why we use GLOBAL lock here. Seems that setQuota does not involve blocks opertions.   Also in `getQuotaUsage`

Reply:
`computeQuotaUsage` needs to rely on block state to get storagespaceConsumed, so the GLOBAL lock is used here. After HDFS-17497 is merged, this global lock can be replaced by the fs lock.

Reply:
@ZanderXu Got it, Thanks sir.